### PR TITLE
Feat/typescript and module fix

### PR DESCRIPTION
**Pull request by dannyhw**
### Issues

fix for partial matches #22
use typescript for validation in development #17 

### changes

I've adjusted the project setup to use typescript and transpile to cjs. This should make it the most compatible possible and improve development experience.

I've also added a fix for the partial matches bug by always adding node_modules/ before the module name given by the user

<details>
  <summary>📦 Published PR as canary version: <code>0.0.13-canary.23.5746ddc.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-react-native-web@0.0.13-canary.23.5746ddc.0
  # or 
  yarn add @storybook/addon-react-native-web@0.0.13-canary.23.5746ddc.0
  ```
</details>

### CODE REVIEW

**Comment by dannyhw** [comment]
@axeldelafosse could you try this canary version and see if it resolves your issue with moti without needing to add node_modules in front of the path?

**Comment by axeldelafosse** [comment]
Yes it's working perfectly 🔥 Thanks !

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "build:watch": "concurrently \"yarn buildBabel:esm -- --watch\" \"yarn buildTsc -- --watch\"",
     "test": "echo \"Error: no test specified\" && exit 1",
     "storybook": "start-storybook -p 6006",
```
Comment (shilman) [inline package.json:35]: Unused?

**Review by shilman — APPROVED** [review]
LGTM. One general q unrelated to this PR is why do we need both `babel` and `webpackFinal` in the preset? They seem redundant?

File: package.json (hunk 1 above)
Comment (dannyhw) [reply package.json:35]: are you refering to `prestorybook`?

I added that because now to run the example the build is needed first.

**Comment by dannyhw** [comment]
> LGTM. One general q unrelated to this PR is why do we need both `babel` and `webpackFinal` in the preset? They seem redundant?

Yeah actually will look into removing that babel part since it does actually seem like its not 100% necessary.

**Comment by dannyhw** [comment]
I've now removed the babel config. it seems that now that the webpack loader now includes the project root the babel config no longer does anything that the loader doesn't do.

**Pull request merged**